subnet/kube/kube.go

```diff
@@ -0,0 +1,292 @@
+	backendTypeAnnotation       = "flannel.alpha.coreos.com/backend-type"
+	backendPublicIPAnnotation   = "flannel.alpha.coreos.com/public-ip"
+
+	cniConfPath       = "/etc/kube-flannel/cni-conf.json"
+	cniConfTargetPath = "/etc/cni/net.d/10-flannel.conf"
+	netConfPath       = "/etc/kube-flannel/net-conf.json"
+)
+
+type kubeSubnetManager struct {
+	client         clientset.Interface
+	nodeName       string
+	nodeStore      cache.StoreToNodeLister
+	nodeController *framework.Controller
+	subnetConf     *subnet.Config
+}
+
+func NewSubnetManager() (subnet.Manager, error) {
+	cfg, err := restclient.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize inclusterconfig: %v", err)
+	}
+	c, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize client: %v", err)
+	}
+
+	podName := os.Getenv("POD_NAME")
+	podNamespace := os.Getenv("POD_NAMESPACE")
+	if podName == "" || podNamespace == "" {
+		return nil, fmt.Errorf("env variables POD_NAME and POD_NAMESPACE must be set")
+	}
+
+	pod, err := c.Pods(podNamespace).Get(podName)
+	nodeName := pod.Spec.NodeName
+	if nodeName == "" {
+		return nil, fmt.Errorf("node name not present in pod spec '%s/%s': %v", podNamespace, podName, err)
+	}
+
+	var cniConf, netConf []byte
+	if _, err := os.Stat(cniConfPath); os.IsNotExist(err) {
+		cniConf = defaultCniConf
+	} else {
+		cniConf, err = ioutil.ReadFile(cniConfPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read cni conf: %v", err)
+		}
+	}
+	netConf, err = ioutil.ReadFile(netConfPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read net conf: %v", err)
+	}
+
+	err = ioutil.WriteFile(cniConfTargetPath, []byte(cniConf), 0644)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write cni conf: %v", err)
+	}
+
+	sc, err := subnet.ParseConfig(string(netConf))
+	if err != nil {
+		return nil, fmt.Errorf("error parsing subnet config: %s", err)
+	}
+	sm, err := newKubeSubnetManager(c, sc, nodeName)
+	if err != nil {
+		return nil, fmt.Errorf("error creating network manager: %s", err)
+	}
+	go sm.Run(context.Background())
+	return sm, err
+}
+
+func newKubeSubnetManager(c clientset.Interface, sc *subnet.Config, nodeName string) (*kubeSubnetManager, error) {
+	var ksm kubeSubnetManager
+	ksm.client = c
+	ksm.nodeName = nodeName
+	ksm.subnetConf = sc
+	ksm.nodeStore.Store, ksm.nodeController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return ksm.client.Core().Nodes().List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return ksm.client.Core().Nodes().Watch(options)
+			},
+		},
+		&api.Node{},
+		resyncPeriod,
+		framework.ResourceEventHandlerFuncs{},
+	)
+	return &ksm, nil
+}
+
+func (ksm *kubeSubnetManager) GetNetworkConfig(ctx context.Context, network string) (*subnet.Config, error) {
+	time.Sleep(time.Second)
```

Comment:
I think we should be able to drop this. Looks to be just init step - and we would only support the default network in this mode.